relayer/src/ethereum_checkpoints/tests/mod.rs

```diff
@@ -1,29 +1,73 @@
-use crate::ethereum_beacon_client::{self, slots_batch, BeaconClient};
+use crate::ethereum_beacon_client::{slots_batch, utils, BeaconClient};
 use checkpoint_light_client::WASM_BINARY;
 use checkpoint_light_client_io::{
     ethereum_common::{
         base_types::BytesFixed,
+        beacon::SyncAggregate,
         network::Network,
         utils::{self as eth_utils, BootstrapResponse, FinalityUpdateResponse, UpdateData},
         SLOTS_PER_EPOCH,
     },
-    replay_back, sync_update,
+    sync_update,
     tree_hash::TreeHash,
     Handle, HandleResult, Init, G2,
 };
 use gclient::{EventListener, EventProcessor, GearApi, Result};
 use parity_scale_codec::{Decode, Encode};
-use tokio::time::{self, Duration};
+use tokio::{
+    sync::{Mutex, MutexGuard},
+    time::{self, Duration},
+};
+
+static LOCK: Mutex<()> = Mutex::const_new(());
 
-const RPC_URL: &str = "http://127.0.0.1:5052";
+const RPC_URL: &str = "http://34.159.93.103:50000";
 
-const FINALITY_UPDATE_5_254_112: &[u8; 4_940] =
-    include_bytes!("./sepolia-finality-update-5_254_112.json");
 const FINALITY_UPDATE_5_263_072: &[u8; 4_941] =
     include_bytes!("./sepolia-finality-update-5_263_072.json");
 const UPDATE_640: &[u8; 57_202] = include_bytes!("./sepolia-update-640.json");
 const BOOTSTRAP_640: &[u8; 54_328] = include_bytes!("./sepolia-bootstrap-640.json");
 
+struct Guard<'a> {
+    _lock: MutexGuard<'a, ()>,
+    pub client: &'a GearApi,
+}
+
+// The struct purpose is to avoid the following error:
+// GearSDK(Subxt(Rpc(ClientError(Call(Custom(ErrorObject { code: ServerError(1014), message: "Priority is too low: (16 vs 16)", data: Some(RawValue("The transaction has too low priority to replace another transaction already in the pool.")) }))))))
+struct NodeClient(GearApi);
+
+impl NodeClient {
+    async fn new() -> Result<Self> {
+        Ok(Self(GearApi::dev().await?))
+    }
+
+    async fn calculate_handle_gas(&self, program_id: [u8; 32], payload: &Handle) -> Result<u64> {
+        Ok(self
+            .0
+            .calculate_handle_gas(None, program_id.into(), payload.encode(), 0, true)
+            .await?
+            .min_limit)
+    }
+
+    async fn lock(&self) -> Guard<'_> {
+        let _lock = LOCK.lock().await;
+
+        Guard {
+            _lock,
+            client: &self.0,
+        }
+    }
+}
+
+#[track_caller]
+fn decode(sync_aggregate: &SyncAggregate) -> G2 {
```

Comment:
Puprose of this `fn` is not clear from the name

Reply:
renamed